compiler/rustc_resolve/src/late.rs

```diff
@@ -2039,6 +2063,43 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
         if let Some(prev_res) = self.r.lifetimes_res_map.insert(id, res) {
             panic!("lifetime {id:?} resolved multiple times ({prev_res:?} before, {res:?} now)")
         }
+
+        match candidate {
+            LifetimeElisionCandidate::Missing(missing @ MissingLifetime { span: elided, .. }) => {
+                debug_assert_eq!(id, missing.id);
+                match res {
+                    LifetimeRes::Static => {
+                        if !self.allow_elided_static {
+                            self.r.lint_buffer.buffer_lint(
+                                lint::builtin::ELIDED_NAMED_LIFETIMES,
+                                missing.id_if_not_fake_or(self.crate_node_id),
```

Comment:
In what cases are the ids fake? In those cases, we should probably just not lint.

Reply:
```rust
pub fn get_mut(&'static self, x: &mut u8) -> &'_ mut u8
```
is ok, but
```rust
pub fn get_mut(&'static self, x: &mut u8) -> &mut u8
```
is not.

Any `MissingLifetime` nodes with `LifetimeKind::Ampersand` get overlooked when we emit buffered lints, for some reason. After staring at the code for a bit, it looked like they get an ad-hoc NodeId not tied to any source code, and not preserved anywhere, because such a lifetime eventually gets replaced with something else.

Not linting here would defeat the whole point of the lint.

Reply:
> Any `MissingLifetime` nodes with `LifetimeKind::Ampersand`

There seem to be more cases beside this one, because I remember getting some more ICEs about not emitting all buffered lints when I ran it on `tests/ui`, but I haven't looked into those yet.